public/docs/_examples/animations/ts/src/hero-list-basic.component.ts

```diff
@@ -0,0 +1,56 @@
+ // #docregion
+ // #docregion imports
+import {
+  Component,
+  Input,
+  animation,
+  state,
+  style,
+  transition,
+  animate
+} from '@angular/core';
+// #enddocregion imports
+import {Hero, Heroes} from './hero.service';
+
+@Component({
+  moduleId: module.id,
+  selector: 'hero-list-basic',
+  // #docregion template
+  template: `
+    <div *ngFor="let hero of heroes.get()"
+         @heroState="hero.state"
+         (click)="hero.toggleActivation()">
+      {{hero.name}}
+    </div>
+  `,
+  // #enddocregion template
+  styleUrls: ['hero-list.component.css'],
+  // #docregion animationdef
+  animations: [
+    animation('heroState', [
+      // Styles in each state
+      state('void',     style({})),
```

Comment:
@matsko The explicit `void` state seems to be necessary because there’s an exception on the initial `void => inactive` transition otherwise. Is it so that we need to have something for all states including `void`, even if we don’t have a transition for it?